frontend/docs/patterns.md

```diff
@@ -194,11 +195,30 @@ const PackComposerPage = ({ router }: IPackComposerPageProps): JSX.Element => {
 export default PackComposerPage;
 ```
 
+
+## Forms
+
+### Data validation
+
+Forms should make use of a pure `validate` function whose input(s) correspond to form data (may include
+new and possibly former form data) and whose output is an object of formFieldName:errorMessage
+key-value pairs (`Record<string,string>`) e.g.
+
+```
+const validate = (newFormData: IFormData) => {
+  const errors = {};
+  ...
+  return errors;
+}
+```
+The output of `validate` should be used by the calling `onChange` handler to set a `formErrors`
```

Comment:
@mike-j-thomas - A little birdy said to me you might be working on more consistent app-wide validation! Just want to confirm with any upcoming product design guidelines for validation before committing to our patterns doc.

Reply:
@RachelElysia thank you for keeping user experience top of mind. We did discuss that this pattern is flexible enough to implement whatever error handling UX we want via logic in the `onChange` and/or `validate` scopes, and is therefore design-agnostic.